test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -651,6 +651,7 @@ public void startup(ICluster cluster)
             }
             catch (Throwable t)
             {
+                startedAt.set(0);
```

Comment:
When the node fails to start setting, leaving that timestamp set is misleading as the node didn't actually start.
What's more important is that it prevents reattempting a startup - as the assertion complains about `startedAt` being non-zero. To workaround it, you need to call `shutdown` which has it's own side-effects - like changing gossip state which isn't desirable by some tests.